### PR TITLE
[YUNIKORN-1446] Adopting table-driven style in the acl unit test

### PR DESCRIPTION
**Pull request by 0yukali0**
### What is this PR for?
Current ACL unit tests do not adopt table-driven style, so I rewrite the ACL unit test file.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1446

### How should this be tested?
make test in the core rep

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/common/security/acl_test.go [hunk 1]
```diff
@@ -19,119 +19,206 @@
...
+	if expected.users != nil && got.users != nil {
+		for username, expectedAllow := range expected.users {
+			if gotAllow, ok := got.users[username]; !ok {
+				return fmt.Errorf("username %s does not exist", username)
+			} else if expectedAllow != gotAllow {
+				return fmt.Errorf("username %s is not same", username)
+			}
+		}
+	} else if (expected.users != nil && got.users == nil) || (expected.users == nil && got.users != nil) {
+		return errors.New("users of an acl is not nil, expect one and got one should be same")
```
Comment (wilfred-s) [inline pkg/common/security/acl_test.go:40]: You are not catching the case that `got` has more users than `expected`
Simplify and change check ordering:
1. check if length of both maps are equal: if not return
2. check equality of the members: let the range handle nil maps. 
3. Combine the user not found and the allow flag difference in one check:
```
gotAllow, ok := got.users[username]; !ok || gotAllow != expectedAllow
```

File: pkg/common/security/acl_test.go [hunk 2]
```diff
@@ -19,119 +19,206 @@
...
 	}
+
+	if expected.groups != nil && got.groups != nil {
```
Comment (wilfred-s) [inline pkg/common/security/acl_test.go:43]: Same as for users

File: pkg/common/security/acl_test.go [hunk 3]
```diff
@@ -19,119 +19,206 @@
...
+		},
+		{
+			"*",
+			UserGroup{User: "user1", Groups: []string{"group1"}},
+			true,
+		},
+		{
+			acl:      "",
+			visitor:  UserGroup{User: "", Groups: nil},
+			expected: false,
```
Comment (wilfred-s) [inline pkg/common/security/acl_test.go:205]: nit: remove struct field names

File: pkg/common/security/acl_test.go [hunk 4]
```diff
@@ -19,119 +19,206 @@
...
+			"",
+			UserGroup{User: "user1", Groups: []string{"group1"}},
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("vistor %v, acl %s", tt.visitor, tt.acl), func(t *testing.T) {
+			acl, err := NewACL(tt.acl)
+			if err != nil {
+				t.Error(err.Error())
```
Comment (wilfred-s) [inline pkg/common/security/acl_test.go:217]: nit: explain that parse error is unexpected as you did before

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: pkg/common/security/acl_test.go (hunk 1 above)
Comment (0yukali0) [reply pkg/common/security/acl_test.go:40]: Hi @wilfred-s 
Understood, thank you for comments.
I am going to update the commit later.

File: pkg/common/security/acl_test.go (hunk 4 above)
Comment (0yukali0) [reply pkg/common/security/acl_test.go:217]: I would remove the err later because the acl string is in the create unit tests.

Comment (0yukali0) [reply pkg/common/security/acl_test.go]: Oops, I forgot that lint would request developer to handle the err in the UT.
So I updated the message of newACL error in the latest commit.

**Review by wilfred-s — APPROVED** [review]
LGTM

**Pull request closed without merging**